core/src/main/java/jenkins/util/AntClassLoader.java

```diff
@@ -919,6 +912,27 @@ public URL getResource(String name) {
         return url;
     }
 
+    /**
+     * Finds a matching file by iterating pathComponents.
+     *
+     * @param pathComponents Path to a folder, split into the individual folder names
+     * @param name File to find
+     * @return Url to found object
+     */
+    @CheckForNull
```

Comment:
Could we use a collection with explicit type in new API? Vector is deprecated

Reply:
@oleg-nenashev, having failed miserably w/ my last couple PR attempts, I'm going to try something new. I've created https://github.com/jsoref/jenkins/pull/1 which probably implements your suggestion. If you like it, I'll merge it into this branch (and thus into this PR).

Reply:
I commented on that PR

Reply:
Would be nice if you request my review there. Might be difficult to get to it soon

Reply:
@oleg-nenashev: it won't let me. You're tagged to it.

It seems like @res0nance is ok with the changes there. I'm unavailable until Thursday, but unless someone else comments on that PR, I'll squash those commits together and merge that commit into this PR (thus there will be two commits in this PR).